docs/ref/forms/models.txt

```diff
@@ -1,15 +1,257 @@
-====================
-Model Form Functions
-====================
-
-Model Form API reference. For introductory material about model forms, see the
-:doc:`/topics/forms/modelforms` topic guide.
+===========
+ModelForms
+===========
 
 .. module:: django.forms.models
-   :synopsis: Django's functions for building model forms and formsets.
+    :synopsis: ModelForm API reference for inner ``Meta`` class and factory
+        functions
+
+.. class:: ModelForm
+
+.. _ref-modelform:
+
+.. currentmodule:: django.forms
+
+:class:`~django.forms.ModelForm` API reference. For introductory material about
+ModelForms, see the :doc:`/topics/forms/modelforms` topic guide.
+
+.. _ref-modelform-meta-options:
+
+ModelForm ``Meta`` Options
+==========================
+
+The :class:`~django.forms.ModelForm` helper class can be used to build forms
+that closely reflect specific Django models. The exact structure of the form
+produced can be controlled by setting metadata options as attributes of an inner
+``Meta`` class.
+
+::
+
+    # Example using some sort of Book model
+
+    from django.forms import ModelForm
+    from myapp.models import Book
+
+
+    class BookForm(ModelForm):
+        class Meta:
+            model = Book
+            fields = ["title", "author"]
+            help_texts = {
+                "title": "The title of the book",
+                "author": "The author of the book",
+            }
+            # ... other attributes
+
+.. admonition:: Invalid Meta class attributes
+
+    While any number of attributes can be included in a ``Meta``
+    class, **only attributes that match those described below will be used**.
+    When troubleshooting or encountering unexpected behavior, be sure to check
+    for typos or misnamed attributes because these will **not**, in and of
+    themselves, raise an error (though they might cause another error to be
+    raised if they affect a required attribute).
+
+:attr:`~ModelFormOptions.model` is required as well as either 
+:attr:`~ModelFormOptions.fields` or :attr:`~ModelFormOptions.exclude`.
+All other ``Meta`` attributes are optional. Included fields need not be 
+included in every option used. If an option isn't set, or if a model's 
+field isn't used with a particular option, Django will fall back on whatever 
+has already been defined for a field in their model or whatever defaults 
+might exist for that field's type.
+
+Optional ``Meta`` attributes, other than
+:attr:`~ModelFormOptions.localized_fields` and
+:attr:`~ModelFormOptions.formfield_callback`, expect a dictionary that maps a
+model field name string to some other object or string. Including invalid or
+excluded field name strings in these dictionaries is possible but should have no
+effect since fields that haven't been included would never be accessed.
```

Comment:
There's bits in the following suggestion you might not like all of it
```suggestion
Required attributes are :attr:`~ModelFormOptions.model`, and either
:attr:`~ModelFormOptions.fields` or :attr:`~ModelFormOptions.exclude`. All
other ``Meta`` attributes are optional.

Optional attributes, other than :attr:`~ModelFormOptions.localized_fields` and
:attr:`~ModelFormOptions.formfield_callback`, expect a dictionary that maps a
model field name to a value. Any field that is not defined in the dictionary
falls back to the field's default value.
```
I also thought maybe we pull out the invalid field names as another admonition
```
.. admonition:: Invalid field names

    An invalid or excluded field name strings in an optional dictionary
    attribute will have no effect since fields that are not included are not
    accessed.
```